samples/net/cloud/aws_mqtt/src/main.c

```diff
@@ -0,0 +1,310 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(aws_mqtt, LOG_LEVEL_DBG);
+
+#include <zephyr.h>
+#include <net/socket.h>
+#include <net/mqtt.h>
+#include <net/net_config.h>
+#include <net/net_event.h>
+#include <net/sntp.h>
+#include <stdio.h>
+#include <sys/printk.h>
+#include <string.h>
+#include <errno.h>
+#include "dhcp.h"
+#include <time.h>
+#include <inttypes.h>
+#include "certificates.h"
+#include "config.h"
+
+#define APP_CA_CERT_TAG 1
+#define APP_PRIVATE_SERVER_KEY_TAG 2
+
+#define RC_STR(rc) ((rc) == 0 ? "OK" : "ERROR")
+#define PRINT_RESULT(func, rc) \
+	LOG_INF("%s: %d <%s>", (func), rc, RC_STR(rc))
+
+static sec_tag_t m_sec_tags[] = {
+    APP_CA_CERT_TAG,
+    APP_PRIVATE_SERVER_KEY_TAG,
+};
+
+
+/* Buffers for MQTT client. */
+static u8_t rx_buffer[APP_MQTT_BUFFER_SIZE];
+static u8_t tx_buffer[APP_MQTT_BUFFER_SIZE];
+static bool connected;
+static u64_t next_alive;
+struct zsock_addrinfo *haddr;
+static struct sockaddr_storage broker;
+static struct mqtt_client client_ctx;
+
+
+static void broker_init(void)
+{
+	struct sockaddr_in *broker4 = (struct sockaddr_in *)&broker;
+
+	broker4->sin_family = AF_INET;
+	broker4->sin_port = htons(CONFIG_AWS_PORT);
+
+	net_ipaddr_copy(&broker4->sin_addr,
+			&net_sin(haddr->ai_addr)->sin_addr);
+
+}
+
+void mqtt_evt_handler(struct mqtt_client *const client,
+		const struct mqtt_evt *evt)
+{
+	int err;
+
+	switch (evt->type) {
+		case MQTT_EVT_CONNACK:
+			if (evt->result != 0) {
+				LOG_ERR("MQTT connect failed %d", evt->result);
+				break;
+			}
+
+			connected = true;
+			LOG_INF("MQTT client connected!");
+			break;
+
+		case MQTT_EVT_DISCONNECT:
+			LOG_INF("MQTT client disconnected %d", evt->result);
+			connected = false;
+			break;
+
+		case MQTT_EVT_PUBACK:
+			if (evt->result != 0) {
+				LOG_ERR("MQTT PUBACK error %d", evt->result);
+				break;
+			}
+
+			LOG_INF("PUBACK packet id: %u", evt->param.puback.message_id);
+
+			break;
+
+		case MQTT_EVT_PUBREC:
+			if (evt->result != 0) {
+				LOG_ERR("MQTT PUBREC error %d", evt->result);
+				break;
+			}
+
+			LOG_INF("PUBREC packet id: %u", evt->param.pubrec.message_id);
+
+			const struct mqtt_pubrel_param rel_param = {
```

Comment:
Do not introduce variables in the middle of the block but place them at the start instead.